cpp/ql/test/library-tests/dataflow/dataflow-tests/test.cpp

```diff
@@ -423,17 +423,17 @@ void intPointerSourceCaller() {
   intPointerSource(&local);
   sink(local); // $ ast,ir=422:7 ast,ir=423:20
 }
-// The IR results for this test _are_ equivalent to the AST ones.
-// The IR annotation is just "ir" because the sink of the unitialized source at
-// `433:7` (i.e., the declaration `int local[1];`) is value of `local`, but the sink
-// of the source from `intPointerSource` value of `*local` (i.e., the indirection node
-// of `local`). So unlike AST dataflow, each of these two sinks correspond to a unique
-// source, and thus we don't need to attach a location annotation to it.
+
+
+
+
+
+
 void intPointerSourceCaller2() {
   int local[1];
   intPointerSource(local);
-  sink(local); // $ ast=433:7 ast=434:20 ir=433:7 ir=434:20 ir
-  sink(*local); // $ ast=433:7 ast=434:20 ir=433:7 ir=434:20
+  sink(local); // $ ast=434:20 ir SPURIOUS: ast=433:7
```

Comment:
I'm not expecting flow here, as discussed at https://github.com/github/codeql/pull/11254#discussion_r1028503418.

Reply:
True. This PR is removing _some_ of the flow, but there's still some that need to be sorted out.

Reply:
Got it. Thanks.

Reply:
Oh, wait. We're getting flow here because of https://github.com/github/codeql/blob/mathiasvp/replace-ast-with-ir-use-usedataflow/cpp/ql/test/library-tests/dataflow/dataflow-tests/test.ql#L84 (and indeed *local is tainted).

Reply:
Ah ok. Then this looks fine.